crates/types/src/net/node.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::BTreeMap;
+
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+use super::TargetName;
+use crate::{cluster::cluster_state::PartitionProcessorStatus, identifiers::PartitionId};
+
+super::define_rpc! {
+    @request=GetNodeState,
+    @response=NodeStateResponse,
+    @request_target=TargetName::NodeGetNodeStateRequest,
+    @response_target=TargetName::NodeGetNodeStateResponse,
+}
```

Comment:
I'm curious about how you (and maybe @AhmedSoliman when available to comment) see this mechanism vs. the new `IdentResponse`? Do we need more than one way to communicate node status info?
 https://github.com/restatedev/restate/blob/18d3b3d7bdf1df546a65e4bf412fa7c49fc2709e/crates/core/protobuf/node_svc.proto#L31-L48

Reply:
Yes, I think there is a difference between the two ways (as far as I understand of course) is that `IdentResposne` is intended for grpc use and hence can be used by external tools like `restatectl` to make call directly to the intended node.

On the other hand GetNodeState is intended for node to node communication and hence can be only called vi the node networking. It can be used for example (as per this PR) by the ClusterController to collect all nodes state to build a view on the status of all nodes. 

Basically all messages that are defined with the `defeine_rpc!` macro is used for cross node communication.

Reply:
A bit slow to come back to this PR - thanks for the answer, this is very helpful context I was missing!